hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/tss/RekeyScenarioOp.java

```diff
@@ -435,7 +437,14 @@ private SpecOperation extractRosterMetadata() {
             activeEntries.set(
                     0, activeEntries.getFirst().copyBuilder().weight(10).build());
             rosterStore.putActiveRoster(new Roster(activeEntries), roundNo + 1);
-            ((CommittableWritableStates) writableStates).commit();
+
+            final var writableTssStates = state.getWritableStates(TssBaseService.NAME);
+            final WritableTssStore writableTssStore = new WritableTssStore(writableTssStates);
+
+            // remove TssEncryptionKeys from state if node ids are not present
+            // in both active and candidate roster's entries
+            writableTssStore.removeIfNotPresent(rosterStore.getCombinedRosterEntriesNodeIds());
+            ((CommittableWritableStates) writableTssStates).commit();
```

Comment:
Having to purge these node IDs in multiple (3?) places feels icky. I wonder if we could encapsulate this functionality inside of the roster store somehow 🤔